test/jit/test_if_hoisting.py

```diff
@@ -0,0 +1,120 @@
+
+import torch
+from torch.testing import FileCheck
+from torch.testing._internal.jit_utils import JitTestCase
+
+if __name__ == "__main__":
+    raise RuntimeError(
+        "This test file is not meant to be run directly, use:\n\n"
+        "\tpython test/test_jit.py TESTNAME\n\n"
+        "instead."
+    )
+
+
+class TestIfHoisting(JitTestCase):
+    def test_if_hoist_basic(self):
+        @torch.jit.script
+        def fn(x: bool, y: int):
+            if x:
+                z = y + 3
+            else:
+                z = y + 3
+            return z
+
+
+        op_graph = fn.graph
+        self.run_pass("common_expression_hoisting", op_graph)
+        self.run_pass("dce", op_graph)
+        FileCheck().check_count("prim::If", 0, exactly=True).run(op_graph)
+        FileCheck().check_count("aten::add", 1, exactly=True).run(op_graph)
+        self.assertEqual(fn(True, 1), 4)
+
+    def test_if_hoist_transposed_expr(self):
+        # Making sure that the if statement
```

Comment:
what are we trying to test here ?

Reply:
oops, the comment is wrong, We are trying to test that the you can still eliminate a line of code if the statement is not already at the front of one of the if blocks.